perceval/backends/gerrit.py

```diff
@@ -69,19 +69,59 @@ def __init__(self, url,
         self.client = GerritClient(self.url, user, max_reviews,
                                    blacklist_reviews)
 
-    @metadata
-    def fetch(self, from_date=DEFAULT_DATETIME):
-        """Fetch the reviews from the repository.
+    def _fetch_gerrit28(self, from_date=DEFAULT_DATETIME):
+        """ Specific fetch for gerrit 2.8 version.
 
-        The method retrieves, from a Gerrit repository, the reviews
-        updated since the given date.
+        Get open and closed reviews in different queries.
+        Take the newer review from both lists and iterate.
+        """
 
-        :param from_date: obtain reviews updated since this date
+        # Convert date to Unix time
+        from_ut = datetime_to_utc(from_date)
+        from_ut = from_ut.timestamp()
 
-        :returns: a generator of reviews
-        """
-        self._purge_cache_queue()
+        filter_open = "status:open"
+        filter_closed = "status:closed"
+
+        last_item_open = self.client.next_retrieve_group_item()
+        last_item_closed = self.client.next_retrieve_group_item()
+        reviews_open = self._get_reviews(last_item_open, filter_open)
+        reviews_closed = self._get_reviews(last_item_closed, filter_closed)
+        last_nreviews_open = len(reviews_open)
+        last_nreviews_closed = len(reviews_closed)
+
+        while reviews_open or reviews_closed:
+            if reviews_open and reviews_closed:
+                if reviews_open[0]['lastUpdated'] >= reviews_closed[0]['lastUpdated']:
+                    review_open = reviews_open.pop(0)
+                    review = review_open
+                else:
+                    review_closed = reviews_closed.pop(0)
+                    review = review_closed
+            elif review_closed and not reviews_open:
```

Comment:
There's a bug here. It should be `reviews_closed` instead of `review_closed`. You can also remove the `and not` condition.